controllers/storageconsumer/storageconsumer_upgrade_controller.go

```diff
@@ -355,3 +400,54 @@ func (r *StorageConsumerUpgradeReconciler) reconcileLocalStorageClient(
 	}
 	return nil
 }
+
+func (r *StorageConsumerUpgradeReconciler) reconcileCephClient(
+	ctx context.Context,
+	name string,
+	namespace string,
+	owner client.Object,
+	additionalOwner client.Object,
+) error {
+	cephClient := &rookCephv1.CephClient{}
+	cephClient.Name = name
+	cephClient.Namespace = namespace
+	if err := r.Client.Get(ctx, client.ObjectKeyFromObject(cephClient), cephClient); err != nil {
+		return err
+	}
+	owners := cephClient.OwnerReferences
+	if idx := slices.IndexFunc(owners, func(ref metav1.OwnerReference) bool {
+		return ref.Kind == "StorageRequest" && ptr.Deref(ref.Controller, false)
+	}); idx != -1 {
+		owners[idx].Controller = nil
+		if err := controllerutil.SetControllerReference(owner, cephClient, r.Scheme); err != nil {
+			return err
+		}
+		if err := controllerutil.SetOwnerReference(additionalOwner, cephClient, r.Scheme); err != nil {
+			return err
+		}
+		util.AddLabel(cephClient, util.CsiCephUserGenerationLabelKey, "0")
+		controllerutil.AddFinalizer(cephClient, util.Pre4_19ProviderCephClientFinalizer)
```

Comment:
Adding a finalizer means that we would need to handle these ceph clients differently during deletion and remove this finalizer

Reply:
Yes.

Reply:
removed.